EmaPajic/assignment4/assignment4.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+@author: EmaPajic
+"""
+
+def count_islands(rows,columns,tiles):
+    """
+    I am assuming that we can change tiles (set some values to false).
+    If we couldn't do that, we could just make another matrix where we would store flags so that we don't visit same tile twice.
+    In this function we are just going through matrix and if we find a part of island we call function find_all_parts_of island to set all parts of that island to false
+    """
+    numOfIslands = 0
+    for i in range(0,rows):
+        for j in range(0,columns):
+            if tiles[i][j] == True:
+                numOfIslands += 1
+                find_all_parts_of_island(rows,columns,i,j,tiles)
+    return numOfIslands
+
+def valid_index(rows,columns,i,j):
+    # check if index is out of range
+    if i < 0 or i >= rows or j < 0 or j >= columns:
```

Comment:
You can also write something like
```
return not (i < 0 or i >= rows or j < 0 or j >= columns)
```
(or transform all conditions to positive)

Reply:
updated :)